### DISCUSSION TITLE
qsub エラー

### DISCUSSION THREAD

**Discussion by Parapriacanthus** [nig-sc/nigsc_homepage2#421] [Q&A] [answered]
qsub を使ってSingularityでバイオインフォのソフトを利用しようとしているのですが、うまくいきません。
singularity 内のコマンドをqsubではなく、直接プロンプトから実行すると、正常に動きます。
BLAST、MiniProtの二つで試しましたが、どちらも上記のように、プロンプトから実行すると作動するが、qsubでは（おそらくsingularity が）うまく働いていないと思います。夏頃は問題なくこれらのプログラムをqsubで実行できていたので、OS移行後の問題かと思います。
どなたか解決方法がわかる方いたら助けてください。

**・エラー内容（ less blstest.e24993682 　の中身）**
/var/spool/age/at083/job_scripts/24993682: 269: cd: can't cd to /var/spool/age/at083/job_scripts/24993682: 370: /opt/pkg/intel/oneapi

**・qsubで投げたスクリプト**

```
#!/bin/bash
#$ -S /bin/sh
#$ -l short
#$ -l s_vmem=20G,mem_req=20G
#$ -cwd
#$ -N MinPro

echo “pwd: $(pwd)”
date
echo starting_job

QUERY=./quey.fasta
DB=/home/user/db/ref.fasta 
FORMAT=6
NCPU=8
EVALUE=1.0e-3
OUTF=test.out


 singularity exec -e /usr/local/biotools/b/blast:2.15.0--pl5321h6f7f691_1 tblastn \
 -query $QUERY \
 -db $DB \
 -outfmt $FORMAT \
 -num_threads $NCPU \
 -evalue $EVALUE \
 -out $OUTF \
 -soft_masking yes \
 -max_target_seqs 100 
 
 
echo ending_job
```

----
（補足）
たとえば、`echo` や `date`などは正常に出力されていてblstest.o24993682 の中身で確認できます。
また、スクリプト内の  singularity exec -e /usr/local/biotools/b/blast:2.15.0--pl5321h6f7f691_1 tblastn 以降を直接プロンプトから実行すると、正常に動きます。

**Comment by sc-fs** [comment, marked as answer]
インタプリタの指定「#$ -S /bin/sh」を「#$ -S /bin/bash」
に変更して実行していただけますか。

**Reply by Parapriacanthus** [reply]
回答ありがとうございます。試したところ問題が解決しました。ありがとうございます。

OS移行とは関係なかったです。
